opentelemetry/proto/profiles/v1/alternatives/arrays/arrays.proto

```diff
@@ -0,0 +1,169 @@
+syntax = "proto3";
+
+package opentelemetry.proto.profiles.v1.alternatives.arrays;
+
+import "opentelemetry/proto/common/v1/common.proto";
+import "opentelemetry/proto/resource/v1/resource.proto";
+
+option go_package = "go.opentelemetry.io/proto/otlp/profiles/v1/arrays";
+
+
+// A pointer from a profile to a trace span.
+message Link {
+  // A unique identifier of a trace that this linked span is part of. The ID is a
+  // 16-byte array.
+  bytes trace_id = 1;
+  // A unique identifier for the linked span. The ID is an 8-byte array.
+  bytes span_id = 2;
+}
+
+// AttributeSet represents a set of attributes. It is referenced from Samples
+// and not embedded because it is common for multiple samples to have the same
+// set of attributes
+message AttributeSet {
+  repeated opentelemetry.proto.common.v1.KeyValue attributes = 1;
+  uint32 dropped_attributes_count = 2;
+}
+
+message Stacktrace {
+  repeated uint32 location_indices = 1;
+}
+
+// message Sample {
+//   uint32 stacktrace_index = 1;
+//   repeated uint32 link_indices = 2;
+//   repeated uint32 attribute_set_indices = 3;
+//   repeated int64 values = 4;
+
+//   // optional
+//   fixed64 timestamp_unix_nano = 5;
+// }
+
+message SampleType {
+  // borrowed from metrics proto
+  enum AggregationTemporality {
+    AGGREGATION_TEMPORALITY_UNSPECIFIED = 0;
+    AGGREGATION_TEMPORALITY_DELTA = 1;
+    AGGREGATION_TEMPORALITY_CUMULATIVE = 2;
+  }
+
+  AggregationTemporality aggregation_temporality = 1;
+  uint64 sample_size = 2;
+
+  // CPU / memory /etc
+  uint32 type_index = 3; // Index into string table.
+  uint32 unit_index = 4; // Index into string table.
+}
+
+message ProfileType {
+  repeated uint32 stacktrace_indices = 10;
+  repeated uint32 link_indices = 11;
+  repeated uint32 attribute_set_indices = 12;
+  repeated int64 values = 13;
+  repeated uint64 timestamps = 14;
+}
+
+message Profile {
+  repeated SampleType sample_types = 1;
+  repeated Stacktrace stacktraces = 2;
```

Comment:
nit: perhaps `stack_traces`? as two words

Reply:
Maybe not. "stacktrace" is fine as a term, I actually like it more. As long as we are consistent in using it as one word :)